### ISSUE TITLE
LND runs out of file descriptors

### ISSUE THREAD

**Issue by apotdevin** [alexbosworth/ln-service#136]
Is there a way to close previous connections that are made? Or close them after the information is received?

Theres this issue that after a while LND runs out of file descriptors and the grpc calls stop working. Referring to this issue https://github.com/apotdevin/thunderhub/issues/172

**Comment by alexbosworth** [comment]
Can you tell me which subscription causes the issue? I can probably fix this if you can figure out which subscription causes it

**Comment by apotdevin** [comment]
ThunderHub is constantly polling `getWalletInfo`, `getClosedChannels`, `getChainBalance`, `getPendingChainBalance` and `getChannelBalance`

Since just leaving thunderhub open increases the file descriptors I'm thinking it must be one of those

**Comment by alexbosworth** [comment]
What about subscriptions?

**Comment by alexbosworth** [comment]
Are you recycling the lnd object or creating new lnd objects all the time?

**Comment by apotdevin** [comment]
Only using `subscribeToInvoice` but this is only called at a specific point. 

Ah good point, it's creating a new lnd object for each call

**Comment by alexbosworth** [comment]
Yes you should re-use the LND object, let me know if that solves it

I will look into `subscribeToInvoice` as well

**Comment by apotdevin** [comment]
Will give it a try, almost sure this is the issue. Thanks for the tip!

**Comment by alexbosworth** [comment]
Another thing if you do recycle the LND, consider that if there is a networking error or something you may want to refresh it

**Comment by apotdevin** [comment]
And is there a correct way to deal with old lnd objects or how are this used connections closed?

**Comment by alexbosworth** [comment]
I'll add a double check on the subscribe to invoice as well so that if you remove the listener it directly cancels the subscription

**Comment by alexbosworth** [comment]
There is now a check on the `subscribeToInvoice` method for the absence of listeners, which should now trigger a discrete cancel on the subscription
